api/scpca_portal/test/factories.py

```diff
@@ -204,3 +206,47 @@ class Meta:
     }
     email = "user@example.com"
     format = DatasetFormats.SINGLE_CELL_EXPERIMENT
+
+
+class JobFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = "scpca_portal.Job"
+        exclude = (
+            "project_id",
+            "sample_id",
+            "download_config_name",
+            "resource_flag",
+            "resource_id",
+        )
+
+    # Test-only fields (not part of the model)
+    project_id = None
+    sample_id = None
+    download_config_name = "MOCK_DOWNLOAD_CONFIG"
+    resource_flag = factory.LazyAttribute(
+        lambda obj: "--project-id" if obj.project_id else "--sample-id"
+    )
+    resource_id = factory.LazyAttribute(
+        lambda obj: obj.project_id if obj.project_id else obj.sample_id
+    )
```

Comment:
Can you think of a way to remove these. I would strongly discourage changing the model we are meant to be testing in order to test. Especially since we define sample_id here then only reference project_id below.

Reply:
I've removed the test-only fields from `JobFactory` and updated the corresponding tests reflect these changes. Thank you for this insight!